### PR TITLE
Update ConfigureAgentLogsUploadToCloudwatch.json

### PR DESCRIPTION
**Pull request by dimaAmz**
Description of changes: 
Updating ConfigureAgentLogsUploadToCloudwatch.json 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by alemartini** [comment]
Hi dimagermanov,
Thank you for submitting a PR.
Could you please describe the changes you have made?
That would help me review your code.

Thanks,

Alessandro

File: Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json [hunk 1]
```diff
@@ -1,257 +1,326 @@
...
+      "description": "(Optional) Enable sharing of logs with CloudWatch of another AWS Account. The ShareDestination parameter needs to be passed if true.",
+      "allowedValues": [
+        "true",
+        "false"
+      ]
+    },
+    "ShareDestination": {
+      "type": "String",
+      "default": "",
+      "description": "(Optional) Destination in the format 'IAMUserAccessKeyID::IAMUserSecretAccessKey::DestinationLogGroup::DestinationLogStream'. The IAM User should have access to DescribeLogStreams, PutLogEvents on the destination CloudWatchLogs resource. The log group and stream should be present. Parameter will be ignored if EnableShare is false",
```
Comment (alemartini) [inline Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json]: We should never pass secrets directly as input parameters.
Use Parameter Store and a SecureString parameter.

File: Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json [hunk 2]
```diff
@@ -1,257 +1,326 @@
...
-        "allowedValues": [
-          "true",
-          "false"
+      "name":"UploadSSMAgentLogsInWindows",
+      "inputs": {
+        "runCommand":[
+          "$loglevel = '{{ LogLevel }}'",
+          "[bool] $enable = ${{ EnableCloudWatchUpload }}",
+          "$logGroup = '{{ LogGroup }}'",
+          "[bool] $enableSharing = ${{ EnableShare }}",
```
Comment (alemartini) [inline Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json]: Do input validation, do not directly use the parameter {{ EnableShare }} as a variable name.
Example:

`
$EnableShareInput = "{{ EnableShare }}"
switch ($EnableShareInput) {
        "true" { $enableSharing = $True }
        "false" { $enableSharing = $False }
        default { throw "Unexpected input."} 
}
`

File: Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json (hunk 1 above)
Comment (dimagermanov) [reply Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json]: Is it better to set default of ShareDestination to parameter from parameter store, or simply fetch it in PowerShell and shell script?

File: Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json (hunk 2 above)
Comment (dimagermanov) [reply Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json]: Yes, having validation sounds good. I'll fix that.

File: Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json (hunk 1 above)
Comment (alemartini) [reply Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json]: The default would be to not share the logs with another account, so you can't set it to a specific parameter name.
I would keep the default value as is (blank), or set it to a placeholder string of your choosing (such as a "DO_NOT_SHARE" string value).
If you use a placeholder string, it's a little easier to implement a flow where you check any passed Parameter is a SecureString before it reaches your Powershell/Bash script. For example:
- Use [aws:branch](https://docs.aws.amazon.com/systems-manager/latest/userguide/automation-branchdocs.html) on the value of {{ ShareDestination }}
  - If {{ ShareDestination }} is not the default placeholder value (i.e. DO_NOT_SHARE), then jump to an aws:assertAwsResourceProperty step, where you want to assert the parameter is of type SecureString. You can call [this API](https://docs.aws.amazon.com/cli/latest/reference/ssm/get-parameter.html) to assert that
  - In all other cases, just continue with the automation without the SecureString check.

**Pull request closed without merging**